### PR TITLE
Rework the dependencies config value

### PR DESCRIPTION
**Pull request by gillchristian**
This PR changes the way dependencies are handled.

```js
// from this
[
  'trae',
  { module: 'ramda', name: 'R' },
  { module: './config', name: 'config', type: 'let' },
  { module: 'es6-promise' }
]

// to this
const trae = require('trae')
const R    = require('ramda')
let config = require('./config')
require('es6-promise')
```

Also added `eslint` since `prettier` only takes care of the formatting.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        "trailingComma": "es5"
+      }
+    ],
+    "no-console": 0
+  },
+  "globals" : {
+    "it": false,
+    "expect": false,
+    "describe": false,
+    "jest": false
```
Comment (fvictorio) [inline .eslintrc.json]: Have you used this? https://www.npmjs.com/package/eslint-plugin-jest

Comment (fvictorio) [reply .eslintrc.json]: Or this: https://stackoverflow.com/a/40265356/3055448

File: e2e/__snapshots__/test.js.snap [hunk 2]
```diff
@@ -105,7 +105,7 @@ ReferenceError: answer is not defined
     at Function.Module._load (module.js:500:3)
     at Function.Module.runMain (module.js:665:10)
     at startup (bootstrap_node.js:187:16)
-    at bootstrap_node.js:608:3
+    at bootstrap_node.js:607:3
```
Comment (fvictorio) [inline e2e/__snapshots__/test.js.snap]: :man_facepalming:

File: e2e/test.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 const fs = require('fs')
 const path = require('path')
 
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (fvictorio) [inline e2e/test.js]: What? `shelljs` should be in the `package.json`

File: src/lib.js [hunk 4]
```diff
@@ -139,7 +141,7 @@ const skip = S.map(R.evolve({ snippets: S.filter(matchNoSkip) }))
 // @link: https://github.com/isaacs/node-glob#glob-primer
 
 // taskOfSnippets :: Object -> Object -> Int -> GlobPattern -> Future([FileResult])
-const taskOfSnippets = (dependencies, globals, timeout, glob) =>
+const taskOfSnippets = ({ dependencies, globals, timeout }, glob) =>
```
Comment (fvictorio) [inline src/lib.js]: Shouldn't `globals` be `config`?

File: src/sanctuary.js [hunk 5]
```diff
@@ -1,9 +1,30 @@
...
+
+const propIsString = prop => R.compose(isString, R.prop(prop))
+const optionalPropIsString = prop =>
+  R.ifElse(R.has(prop), propIsString(prop), R.T)
+
+const checkProps = R.allPass([
+  propIsString('module'),
+  optionalPropIsString('name'),
+  optionalPropIsString('type'),
+])
```
Comment (fvictorio) [inline src/sanctuary.js]: Nice.

File: .eslintrc.json (hunk 1 above)
Comment (gillchristian) [reply .eslintrc.json]: Didn't know about them :man_facepalming: 

Will use the latter

File: src/lib.js (hunk 4 above)
Comment (gillchristian) [reply src/lib.js]: The whole object is config, I'm just destructing.

Comment (gillchristian) [reply src/lib.js]: Yeah it should, don't know why I added that instead of installing. Will fix

Comment (fvictorio) [reply src/lib.js]: Ohhh, you sleazy destructurer.

**Pull request merged**